File: app/src/main/java/ar/com/wolox/android/example/ui/viewpager/ViewPagerActivityModule.java
package ar.com.wolox.android.example.ui.viewpager;

import android.app.Activity;

import ar.com.wolox.android.example.ui.viewpager.random.RandomFragment;
import ar.com.wolox.android.example.ui.viewpager.request.RequestFragment;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.android.ContributesAndroidInjector;
import dagger.multibindings.IntoMap;

@Module(subcomponents = ViewPagerActivitySubcomponent.class)
public abstract class ViewPagerActivityModule {

    @Binds
    @IntoMap
    @ActivityKey(ViewpagerActivity.class)
    abstract AndroidInjector.Factory<? extends Activity> bindViewPagerActivityFactory(ViewPagerActivitySubcomponent.Builder builder);

    @ContributesAndroidInjector
    abstract RandomFragment randomFragment();

    @ContributesAndroidInjector
    abstract RequestFragment requestFragment();

}

File: app/src/main/java/ar/com/wolox/android/example/ui/viewpager/ViewPagerActivityModule.kt
package ar.com.wolox.android.example.ui.viewpager

import ar.com.wolox.android.example.ui.viewpager.fragment.ViewPagerFragment

import ar.com.wolox.android.example.ui.viewpager.random.RandomFragment
import ar.com.wolox.android.example.ui.viewpager.request.RequestFragment

import dagger.Binds
import dagger.Module
import dagger.android.AndroidInjector
import dagger.android.ContributesAndroidInjector
import dagger.multibindings.ClassKey
import dagger.multibindings.IntoMap

@Module(subcomponents = [(ViewPagerActivitySubcomponent::class)])
abstract class ViewPagerActivityModule {

    @Binds
    @IntoMap
    @ClassKey(ViewPagerActivity::class)
    internal abstract fun bindViewPagerActivityFactory(
        builder: ViewPagerActivitySubcomponent.Builder
    ): AndroidInjector.Factory<*>

    @ContributesAndroidInjector
    internal abstract fun viewpagerFragment(): ViewPagerFragment

    @ContributesAndroidInjector
    internal abstract fun randomFragment(): RandomFragment

    @ContributesAndroidInjector
    internal abstract fun requestFragment(): RequestFragment
}
